src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  static removeDeadAnimals = () => {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  };
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  bite(target) {
+    if (target instanceof Herbivore && !target.hidden) {
+      target.health -= 50;
+      Animal.removeDeadAnimals();
```

Comment:
Instead of calling `Animal.removeDeadAnimals()` here, you could filter the `Animal.alive` array directly after the health deduction to remove any dead animals. This would be more efficient and keep the logic contained within the `bite` method.